src/neptune/internal/operation_processors/async_operation_processor.py

```diff
@@ -162,13 +162,20 @@ def wait(self) -> None:
     def _check_queue_size(self) -> bool:
         return self._queue.size() > self._batch_size / 2
 
-    def _wait_for_queue_empty(self, initial_queue_size: int, seconds: Optional[float]) -> None:
+    def _wait_for_queue_empty(
+        self,
+        initial_queue_size: int,
+        seconds: Optional[float],
+        msg_queue: Optional[Queue] = None,
```

Comment:
It's probably better idea to use some dataclass that will b sent over the queue to different in-between values vs final vs `None` etc.